exec.go

```diff
@@ -979,15 +977,26 @@ func runproc(win *Window, s string, rdir string, newns bool, argaddr string, arg
 		if err == nil {
 			if cpid != nil {
 				cpid <- cmd.Process
+			} else {
+				cpid <- nil
 			}
-			return // TODO(flux) where do we wait?
+			go func() {
+				cmd.Wait()
+				Closeall()
+				cwait <- cmd.ProcessState
```

Comment:
Maybe the right fix for the fid closing would seem to be to snapshot the selection on command start (it does already in `Window.wrselrange`) but update it here only if the user hasn't updated it in the intervening time?

Reply:
I don't think that's necessary if we pass `fid` directly to `exec.Cmd`. See my comment above.

Reply:
I agree that my comment above "right fix for fid closing" is not necessary in this PR. This was largely musing on my part. There is however a UX problem with Edwood:

* have a `Window` with a selection
* middle click a `|command`
* the command takes a long time to run
* type in the window in the intervening time before `command` finishes

This may fail (and trigger a `panic`) because fid closing assumes that `t.cache` is empty.